frontend/rust-lib/flowy-database2/src/services/calculations/controller.rs

```diff
@@ -215,19 +217,34 @@ impl CalculationsController {
     }
   }
 
-  pub async fn did_receive_row_changed(&self, row: Row) {
+  pub async fn did_receive_row_changed(&self, row: Row, view_id: Option<String>) {
     self
       .gen_task(
-        CalculationEvent::RowChanged(row),
+        CalculationEvent::RowChanged(row, view_id),
         QualityOfService::UserInteractive,
       )
       .await
   }
 
-  async fn handle_row_changed(&self, row: Row) {
+  async fn handle_row_changed(&self, row: Row, view_id: Option<String>) {
     let cells = row.cells.iter();
     let mut updates = vec![];
 
+    // In case there are calculations where empty cells are counted
+    // as a contribution to the value.
+    if cells.len() == 0 {
+      if let Some(view_id) = view_id {
+        let calculations = self.delegate.get_all_calculations(&view_id).await;
+        for calculation in calculations.iter() {
+          let update = self.get_updated_calculation(calculation.clone()).await;
+          if let Some(update) = update {
+            updates.push(CalculationPB::from(&update));
+            self.delegate.update_calculation(&view_id, update);
+          }
+        }
+      }
```

Comment:
The calculation controller lives in the context of a view editor. Is there another reason why Self::view_id isn't used here?